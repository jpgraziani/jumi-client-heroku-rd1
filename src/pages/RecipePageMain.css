
.single-reicpe {
  padding: 5rem 0 0 0;
}
.single-recipe-info {
  width: 80vw;
  display: grid;
  grid-template-columns: 1fr;
  margin: 2rem auto;
}
.desc,
.info {
  margin: 1rem 0;
}
.desc h3 {
  text-transform: capitalize;
  letter-spacing: var(--mainSpacing);
}
.desc p {
  line-height: 1.5;
}
.info h3,
.info h6 {
  text-transform: capitalize;
  letter-spacing: var(--mainSpacing);
}

.info h6 {
  font-weight: 300;
}
.recipe-ingredients {
  width: 80vw;
  margin: 0 auto 3rem auto;
}
.recipe-ingredients h6 {
  text-transform: capitalize;
  letter-spacing: var(--mainSpacing);
}
.ingredients, .directions {
  list-style-type: none;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(235px, 1fr));
  grid-column-gap: 2rem;
  grid-row-gap: 1rem;
}
.ingredients ul {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.directions {
  grid-template-columns: repeat(auto-fit, minmax(330px, 1fr));
}

.health-info {
  display: flex;
  justify-content: space-around;
  text-align: center;
}

.health-info h6 {
  margin-bottom: .25rem;
  color: var(--lightGreen);
}

@media screen and (min-width: 992px) {
  .single-recipe-info,
  .recipe-ingredients {
    width: 95vw;
    max-width: 1170px;
  }
  .single-recipe-info {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 2rem;
  }
  .info {
    padding-left: 3rem;
  }
}
