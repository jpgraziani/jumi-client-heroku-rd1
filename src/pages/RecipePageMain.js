import React from 'react'
import { Link } from 'react-router-dom'
import Hero from '../components/Hero'
import Banner from '../components/Banner'
import ApiContext from '../ApiContext'
import { findRecipe } from '../recipe-helpers'
import './RecipePageMain.css'

export default class RecipePageMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = ApiContext


  render() {
    const { recipes=[] } = this.context
    const { recipeId } = this.props.match.params
    const recipe = findRecipe(recipes, recipeId) || { content: '' }
    return (
      <>
        <Hero hero='defaultHero'>
          <Banner title={`${recipe.name}`}>
            <Link to='/recipes' className='btn-primary'>
              back to recipes
            </Link>
          </Banner>
        </Hero>
        <section className='single-recipe'>
          <div className='single-recipe-info'>
            <article className='desc'>
              <h3>ingredients</h3>
              <hr />
              {recipe.ingredients}
            </article>
          <article className='desc'>
            <h3>directions</h3>
            <hr />
              {recipe.directions}
            </article>
            <article className='desc health-hr'>
              <h3>health benefits</h3>
              <hr />
              <article className='health-info'>
                <div>
                  <h6>calories</h6>
                  {recipe.calories}
                </div>
                <hr className='hrVertical' />
                <div>
                  <h6>protein</h6>
                  {recipe.protein}
                </div>
              </article>
              <hr className='hrThin' />
            </article>
          </div>
        </section>
      </>
    )
  }
}
