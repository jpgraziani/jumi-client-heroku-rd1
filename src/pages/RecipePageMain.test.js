import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RecipePageMain from './RecipePageMain'


describe(`RecipePageMain component`, () => {
  it('renders a .RecipePageMain by default', () => {
    const wrapper = shallow(<RecipePageMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})