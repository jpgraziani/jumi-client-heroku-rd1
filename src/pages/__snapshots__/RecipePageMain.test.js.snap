// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePageMain component renders a .RecipePageMain by default 1`] = `
<Fragment>
  <Hero
    hero="defaultHero"
  >
    <Banner
      title="undefined"
    >
      <Link
        className="btn-primary"
        replace={false}
        to="/recipes"
      >
        back to recipes
      </Link>
    </Banner>
  </Hero>
  <section
    className="single-recipe"
  >
    <div
      className="single-recipe-info"
    >
      <article
        className="desc"
      >
        <h3>
          ingredients
        </h3>
        <hr />
      </article>
      <article
        className="desc"
      >
        <h3>
          directions
        </h3>
        <hr />
      </article>
      <article
        className="desc health-hr"
      >
        <h3>
          health benefits
        </h3>
        <hr />
        <article
          className="health-info"
        >
          <div>
            <h6>
              calories
            </h6>
          </div>
          <hr
            className="hrVertical"
          />
          <div>
            <h6>
              protein
            </h6>
          </div>
        </article>
        <hr
          className="hrThin"
        />
      </article>
    </div>
  </section>
</Fragment>
`;
